packages/runtime-core/__tests__/apiCreateApp.spec.ts

```diff
@@ -362,6 +363,30 @@ describe('api: createApp', () => {
     expect(handler).toHaveBeenCalledTimes(1)
   })
 
+  describe('instance returned by the mount function', () => {
+    test('return public instance when stateful component is the root of the app', () => {
+      const app = createApp({ render: () => h('h1') })
+      expect(app.mount(nodeOps.createElement('div'))).toBeDefined()
+    })
+
+    test('return null when functional component is the root of the app', () => {
+      const app = createApp(() => h('h1'))
+      expect(app.mount(nodeOps.createElement('div'))).toBe(null)
+    })
+
+    test('return null when async component wrapper is the root of the app', () => {
+      const app = createApp(
+        defineAsyncComponent(
```

Comment:
I don't think we should support an async component as the root because the point of an async component is to delay its fetching to when it's used but the root component is always rendered so it would always be fetched

Reply:
If so, we need to print the appropriate warning message

Reply:
We could add a warning message but I wonder how often someone will use an async component as the root component.

Reply:
I agree that almost no one will do this, but as you can see, someone has submitted a related issue, so we can’t ignore it ;)

Reply:
I think we can wait to see if it happens more often before adding a warning. After all, one case is not enough to justify a warning

Reply:
> I don't think we should support an async component as the root

Is it too big a change for that ? Most of the changes I'm seeing here are about tests. The actual code change seems pretty minimal, and then users can expect async components to work everywhere components are expected.

What are the disadvantages of returning null when the root component is async ? Most of the Vue code I ever saw never uses the root public instance for anything...